src/arrayMethodJoin.js

```diff
@@ -5,7 +5,28 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let resultStr = '';
+    let normalizedSeparator;
+
+    if (separator === undefined) {
+      normalizedSeparator = ',';
+    } else {
+      normalizedSeparator = String(separator);
+    }
+
+    for (let value of this) {
+      if (value === null || value === undefined) {
+        value = '';
+      }
+
+      resultStr += String(value) + normalizedSeparator;
+    }
+
+    if (normalizedSeparator === '') {
+      return resultStr;
+    }
+
+    return resultStr.slice(0, -normalizedSeparator.length);
```

Comment:
this is an unnecessary complication - please rethink your solution without the `slice` method

you can create your `resultStr` without this code-line by adding some extra condition to your previous code and return just `resultStr`